zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -792,7 +804,7 @@ protected void syncWithLeader(long newLeaderZxid) throws Exception {
             // Similar to follower, we need to log requests between the snapshot
             // and UPTODATE
             ObserverZooKeeperServer ozk = (ObserverZooKeeperServer) zk;
-            for (PacketInFlight p : packetsNotCommitted) {
+            for (PacketInFlight p : packetsNotLogged) {
```

Comment:
it's not completely obvious which list to use here. i think a good comment on the semantics of the two lists is key.